### PR TITLE
aries_mmr_configurable sampler

### PR DESCRIPTION
**Pull request by oceandlr**
Aries_mmr_configrurable sampler:  sets/resets/reads configurable Aries counters.

### CODE REVIEW

File: ldms/src/sampler/aries_mmr/Plugin_aries_mmr.man [hunk 1]
```diff
@@ -14,22 +14,21 @@ config name=aries_nic_mmr [ <attr> = <value> ]
...
 .PP
-The aries_XXX_mmr samplers depend on Cray's libgpcd, built with aries options. This library is currently
-unreleased and you cannot build this sampler if you do not have the libraries and headers.
+The aries_XXX_mmr samplers depend on Cray's libgpcd, built with aries options. This library has been released by Cray in CLE6 and later. You cannot build this sampler if you do not have the libraries and headers.
 If you have the code to build the library, be sure to build with
 .B CFLAGS=-fPIC
 
 .PP
-The difference between the variants is that aries_nic_mmr will skip any counters in the inputfile
+The difference between the ariants is that aries_nic_mmr will skip any counters in the inputfile
```
Comment (tom95858) [inline ldms/src/sampler/aries_mmr/Plugin_aries_mmr.man:30]: typo?

File: ldms/src/sampler/aries_mmr/check_mmr_configurable.c [hunk 2]
```diff
@@ -0,0 +1,221 @@
+#include <inttypes.h>
+#include <errno.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include "gpcd_lib.h"
+#include "gpcd_pub.h"
+
+//TOBUILD:  gcc -L /opt/cray/gni/default/lib64 -lgpcd -I/opt/cray/gni/default/include/gpcd check_mmr_configurable.c
```
Comment (tom95858) [inline ldms/src/sampler/aries_mmr/check_mmr_configurable.c:9]: C++ comment style?

File: ldms/src/sampler/aries_mmr/check_mmr_configurable.c [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+		exit(-1);
+
+ 	}
+
+	printf("Initial read_mmr_val '%s' = %" PRIu64 "\n", met, temp);
+
+
+	//	temp = 111100001111;
+	temp = 666;
+//	temp = 0;
```
Comment (tom95858) [inline ldms/src/sampler/aries_mmr/check_mmr_configurable.c:164]: Remove this?

File: ldms/src/sampler/aries_mmr/Plugin_aries_mmr.man (hunk 1 above)
Comment (oceandlr) [reply ldms/src/sampler/aries_mmr/Plugin_aries_mmr.man:30]: yep, thanks

File: ldms/src/sampler/aries_mmr/check_mmr_configurable.c (hunk 2 above)
Comment (oceandlr) [reply ldms/src/sampler/aries_mmr/check_mmr_configurable.c:9]: ok

File: ldms/src/sampler/aries_mmr/check_mmr_configurable.c (hunk 3 above)
Comment (oceandlr) [reply ldms/src/sampler/aries_mmr/check_mmr_configurable.c:164]: ok

**Comment by oceandlr** [comment]
Going to do a review with @bschwal . Will note here when ready.

**Comment by oceandlr** [comment]
This has been revised to specify counters as either U64 or hex in the set and also in the read.

**Comment by oceandlr** [comment]
@tom95858 can this be taken in?

**Pull request merged**